### ISSUE TITLE
Duplicate compile class name when use @unocss/transformer-compile-class

### ISSUE THREAD

**Issue by nguyentuevuong** [unocss/unocss#2667]
### UnoCSS version

0.52.3

### Describe the bug

When using the uno 0.52.3 version, if you use the transformer class with a non-default classPrefix configuration for vite and vuejs, you will encounter the error shown in the image below:

![image](https://github.com/unocss/unocss/assets/23444528/50c7d48a-da55-4def-bf9e-26ec9d18b424)

How to use:
![image](https://github.com/unocss/unocss/assets/23444528/61246a3b-2219-4c73-bf0e-8fd6a2d5c0fd)

uno.config.ts configuration file:

```ts
import {
  defineConfig, 
  presetUno, 
  transformerCompileClass,
} from 'unocss';

export default defineConfig({
  presets: [
    presetUno(),
  ],
  include: /(vue|ts|html)/,
  exclude: /(node_modules)/,
  transformers: [
    transformerCompileClass({
     // This configuration is causing an error.
      classPrefix: 'ui-',
    }),
  ],
});
```

### Reproduction

StackBlitz sample:
https://stackblitz.com/edit/vitejs-vite-r1w7qg?file=uno.config.ts

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/unocss/unocss/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't [already an issue](https://github.com/unocss/unocss/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/unocss/unocss/discussions) or join our [Discord Chat Server](https://chat.antfu.me).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by dycops** [comment]
Have a solution?

**Comment by nguyentuevuong** [comment]
> Have a solution?

You can temporarily use the solution I implemented in this PR #2664, as the described issues have been resolved by me. However, the PR has not been approved yet.

**Comment by jacob-8** [comment]
> if you use the transformer class with a non-default classPrefix

This error is also showing up with the default `:uno:` prefix as seen [here](https://stackblitz.com/fork/github/unocss/unocss/tree/stackblitz-examples/examples/astro-vue)
